Content.Server/Ninja/Systems/NinjaGlovesSystem.cs

```diff
@@ -0,0 +1,91 @@
+using Content.Server.Communications;
+using Content.Server.DoAfter;
+using Content.Server.Ninja.Systems;
+using Content.Server.Power.Components;
+using Content.Shared.DoAfter;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Ninja.Components;
+using Content.Shared.Ninja.Systems;
+using Content.Shared.Popups;
+using Content.Shared.Research.Components;
+
+namespace Content.Server.Ninja.Systems;
+
+public sealed class NinjaGlovesSystem : SharedNinjaGlovesSystem
+{
+    [Dependency] private readonly NinjaSystem _ninja = default!;
+
+    protected override void OnDrain(EntityUid uid, NinjaDrainComponent comp, InteractionAttemptEvent args)
```

Comment:
I think each method in this file could do with a comment to explain what you're trying to achieve. I can see that the gloves are used to advance objectives like steal research. A one or two liner to spell that out would be helpful.